app/src/main/java/org/liberty/android/fantastischmemo/ui/RecentListFragment.java

```diff
@@ -303,24 +325,29 @@ public void onBindViewHolder(final ViewHolder holder, final int position) {
             holder.itemView.setOnClickListener(new View.OnClickListener() {
                 @Override
                 public void onClick(View v) {
-                    Intent myIntent = new Intent();
-                    myIntent.setClass(context, StudyActivity.class);
-                    String dbPath = currentItem.dbPath;
-                    myIntent.putExtra(StudyActivity.EXTRA_DBPATH, dbPath);
-                    recentListUtil.addToRecentList(dbPath);
-                    context.startActivity(myIntent);
+                    if (getSelectedItemCount() == 0) {
+                        Intent myIntent = new Intent();
+                        myIntent.setClass(context, StudyActivity.class);
+                        String dbPath = currentItem.dbPath;
+                        myIntent.putExtra(StudyActivity.EXTRA_DBPATH, dbPath);
+                        recentListUtil.addToRecentList(dbPath);
+                        context.startActivity(myIntent);
+                    } else {
+                        toggleSelection(position);
+                    }
+                    updateActionMode();
                 }
             });
 
             holder.itemView.setOnLongClickListener(new View.OnLongClickListener() {
                 @Override
                 public boolean onLongClick(View v) {
-                    String dbPath = currentItem.dbPath;
-                    DialogFragment df = new OpenActionsFragment();
-                    Bundle b = new Bundle();
-                    b.putString(OpenActionsFragment.EXTRA_DBPATH, dbPath);
-                    df.setArguments(b);
-                    df.show(((FragmentActivity)context).getSupportFragmentManager(), "OpenActions");
+                    AppCompatActivity appCompatActivity = (AppCompatActivity) context;
```

Comment:
Your concern is right. Converting context to the activity is not the right approach. There are two ways to deal with this problem:
1. In holder's constructor, you need to set a callback Predicate<Void> startSupportActionModeProxy
When you create the Adapter, you will need to pass in the startSupportActionModeProxy from Activity thus passing down all the way to holder.
Then you can create the startSupportActionModeProxy in Activity with startSupportActionMode(actionModeCallback)

2. A simpler and less verbose way is using LocalBroadcastManager, when you want to startSupportActionMode, you basically send a broadcast intent. In the Activity, you register the broadcastReceiver in onStart and unregister in onStop and listen to the intent action you defined. When you receive the intent to startSupportActionMode, just call startSupportActionMode(actionModeCallback)

3. If you want to use EventBus, you will need to add the dependency: http://greenrobot.org/eventbus/, and this is by far the simplest, but it requires you to add a library.
It is very similar to approach 2.